### PR TITLE
[WIP] Cmake

### PR DESCRIPTION
**Pull request by mjvk**
Some todos:

- [ ] Wait until  USNavalResearchLaboratory/protolib#16 is in.
- [ ] Fix some unintentional changes (unixPostProcess.cpp, some examples).
- [ ] Discuss what cmake version is acceptable (FetchContent is only >= 3.11).
- [ ] Test WAF and make to see if I did not break anything.
- [ ] Test more platforms

### CODE REVIEW

**Comment by NWittje** [comment]
I am using Norm in a separate project with cross linux/windows usage, so CMake integration greatly appreciated!

File: src/common/normSession.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #include <time.h>  // for gmtime() in NormTrace()
 
 #include "protoPktETH.h"
-#include "protoPktIP.h"
+#include "protoPktIp.h"
```
Comment (bebopagogo) [inline src/common/normSession.cpp:10]: I'm not sure this change is correct?  The "ProtoPktIP" header file is named with uppercase "IP" per the original code.

File: src/common/pcap2norm.cpp [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #include <pcap.h>
 #include <sys/socket.h>  // for PF_ types (protocol family)
 #include "protoPktETH.h" // for Ethernet frame parsing
-#include "protoPktIP.h"  // for IP packet parsing
+#include "protoPktIp.h"  // for IP packet parsing
```
Comment (bebopagogo) [inline src/common/pcap2norm.cpp:10]: Same here as other comment.

File: src/common/normSession.cpp (hunk 1 above)
Comment (mjvk) [reply src/common/normSession.cpp:10]: Fixed

File: src/common/pcap2norm.cpp (hunk 2 above)
Comment (mjvk) [reply src/common/pcap2norm.cpp]: Fixed

**Comment by b00ga** [comment]
I don’t quite have it working yet, but I’ve been working on another NRL code base using ExternalProject. I think this is the right way to include other sources, as opposed to FetchContent.
https://cmake.org/cmake/help/latest/module/ExternalProject.html

**Comment by mjvk** [comment]
FetchContent is the configure time version of external project
. Imho quite a bit nicer. 

Do you have a protolib somewhere else on your system. Cmake might pick that one up instead of using FetchContent

**Comment by b00ga** [comment]
Here's a comparison of CMake versions across various platforms/repositories:
https://gist.github.com/b00ga/21b3edf4de1805d098d1bad9af21d4b9#file-cmake-versions-md

3.11 minimum means bringing your own CMake to RHEL/CentOS 6, Ubuntu 16.04 LTS, Ubuntu 18.04 LTS, and Debian 9.

**Pull request merged**